### PR TITLE
CORTX-29037 : Remove 'pytest' dependency

### PR DESCRIPTION
**Pull request by akash2144**
Signed-off-by: Akash Dudhane <akash.a.dudhane@seagate.com>

# Problem Statement
-  CORTX-29037 : Remove 'pytest' dependency

# Design
-  cortx-ha granular image creation is failing due to newly added package `pytest`.  As per new HA test framework, pytest package is not required so removing pytest package from ha requirements ( for creation of granular image )

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: N
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N] NA
- [ ] Side effects on other features (deployment/upgrade)? [Y/N] N
- [ ] Dependencies on other component(s)? [Y/N] N
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

# Test Details:
Note :  Most of the testcases are failing due to unknown reason ( either it's a legacy code Or design changes )
 
[results_with_pytest.txt](https://github.com/Seagate/cortx-ha/files/8390289/results_with_pytest.txt)
[results_with_unittest.txt](https://github.com/Seagate/cortx-ha/files/8390295/results_with_unittest.txt)

### CODE REVIEW

**Review by mukhtar-inamdar — APPROVED** [review]

**Review by Madhura-08 — APPROVED** [review]

**Review by gauri-bhosale — APPROVED** [review]

**Review by mariyappanp — APPROVED** [review]

**Comment by ArchanaLimaye** [comment]
1. Please confirm that those tests that are passing ; keep on passing after the pytest package is removed 
2. Confirm that HA RPM is successfully generated and can be deployed successfully after this change

Update both outputs in test details 
@akash2144

File: ha/test/main.py [hunk 1]
```diff
@@ -25,9 +25,12 @@
 import os
 import pathlib
 import sys
-import pytest
+import unittest
```
Comment (ArchanaLimaye) [inline ha/test/main.py:28]: pytest package is replaced with unittest right ? are any issues seen in installation of this unititest ?

Comment (akash2144) [reply ha/test/main.py:28]: 'unittest' comes packaged with Python, no need to install it separately.

**Comment by akash2144** [comment]
>> 1. Please confirm that those tests that are passing ; keep on passing after the pytest package is removed

_with Unittest results are :_
	Ran 17 tests in 47.411s
	FAILED (errors=8)
_with pytest results are :_
	15 warnings, 1 error in 1.05s

I don't think pytest was in use, as per results it's showing only 1 error (i.e. in test_cluster_validation.py file) But as per our latest code changes we expect additional errors in :
test_hw_resource.py = No such file or directory: '/etc/cortx/ha/decision_monitor_conf.json'
test_consul_kv.py = Failed to establish a new connection (trying to connect localhost instead od consul server)
test_cluster_stop = haclient group is not present on the system
etc.
>> 2. Confirm that HA RPM is successfully generated and can be deployed successfully after this change

I have successfully created rpm & installed it in container. ( Test directory is not part of rpm creation so it will not affect rpm creation/installation )

**Pull request merged**